doc/main/usage.rst

```diff
@@ -186,6 +186,63 @@ Output
 The ROOT "MC truth" output file, if enabled with the command above, contains
 hits from all the sensitive detectors.
 
+
+.. _celer-geo:
+
+Visualization application (celer-geo)
+=====================================
+
+The ``celer-geo`` app is a server-like front end to the Celeritas geometry
+interfaces that can generate exact images of a user geometry model.
+
+Usage::
+
+  celer-geo {input}.jsonl
+            -
+
+Input
+-----
+
+The input and output are both formatted as `JSON lines`_. A single input JSON
+on a single line executes a command in ``celer-geo``, and a single JSON line is
+printed to ``stdout``. Log messages are sent to ``stderr`` and can be
+controlled by the :ref:`environment` variables.
+
+The first input command must define the input model (and may define additional
+device settings)::
+
+   {"geometry_file": "simple-cms.gdml"}
+
+Subsequent lines will each specify the imaging window, the geometry, the
+binary image output filename, and the execution space (device or host for GPU
+or CPU, respectively).::
+
+   {"image": {"_units": "cgs", "lower_left": [-800, 0, -1500], "upper_right": [800, 0, 1600], "rightward": [1, 0, 0], "vertical_pixels": 128}, "volumes": true, "bin_file": "simple-cms-cpu.orange.bin"}
+
+After the first image window is specified, it will be reused if the "image" key
+is omitted. A new geometry and/or memory space may be specified, useful for
+verifying different navigators behave identically::
+
+   {"bin_file": "simple-cms-cpu.geant4.bin", "geometry": "geant4"}
```

Comment:
So this is just a binary dump of the raytrace data --- what will be your approach to converting this into a image?

Reply:
The python visualization script (waiting for a subsequent PR, see https://github.com/sethrj/celeritas/commit/92ec52360838ffef2e03609252d0b51cffd3f80f#diff-6b30a61667ce57aee53738523a91f9f1bb5c87901132689130e4e4e883df7c7a ) reads in the binary file,  maps the integers to colors, and renders the image with matplotlib.

Reply:
I think at some point we'll want to take an approach closer to the openmc visualization tool: take an ID->color mapping from the user, generate an actual image on the GPU (or host), and send either the PNG or (if `libpng` isn't installed) the uncompressed image back to the host.

Reply:
Okay we will want to make sure that this the visualization script is mentioned here (in the next PR)